### PR TITLE
chore: add auto labeler workflow 

### PR DESCRIPTION
**Pull request by hexqi**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution

添加workflow：提交PR或者修改标题后自动添加/更新 label
自动添加的label包括：
- enhancement：标题以feat: 开头
- document: 标题以docs: 开头
- bug: 标题以fix:开头
- refactor-main: 目标分支为refactor/develop
- ospp-2024: 目标分支为 ospp-2024/*

### What is the current behavior?
验证：
image

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced an automated labeling system for pull requests based on commit message patterns, enhancing project management.
	- Added a GitHub Actions workflow to automatically apply labels when pull requests are opened or edited, improving organization.

- **Documentation**
	- Configured labels for enhancements, documentation, bugs, and specific projects, streamlining the contribution process.

### CODE REVIEW

File: .github/workflows/auto-labeler.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+on:
+  pull_request_target:
+    types: [opened, edited]
+
+permissions:
+  # Setting up permissions in the workflow to limit the scope of what it can do. Optional!
+  contents: read # the config file
+  issues: write # for labeling issues (on: issues)
+  pull-requests: write # for labeling pull requests (on: pull_request_target or on: pull_request)
+  statuses: write # to generate status
```
Comment (chilingling) [inline .github/workflows/auto-labeler.yml]: `statuses`  和 `checks` 权限好像都没必要开

File: .github/workflows/auto-labeler.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  checks: write # to generate status
+
+jobs:
+  label:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (chilingling) [inline .github/workflows/auto-labeler.yml]: 这里好像只需要读取 PR 的相关信息，不需要检出到对应的分支？
如果需要，可以切到 `v4`

File: .github/auto-labeler.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+version: v1
```
Comment (chilingling) [inline .github/auto-labeler.yml:1]: 最新版本应该是到 `v4` 了，可以升级一下
image

File: .github/workflows/auto-labeler.yml (hunk 1 above)
Comment (hexqi) [reply .github/workflows/auto-labeler.yml]: 这个action有提供这块的功能，暂时没有用到，可以去掉

File: .github/workflows/auto-labeler.yml (hunk 2 above)
Comment (hexqi) [reply .github/workflows/auto-labeler.yml]: 已删除

File: .github/auto-labeler.yml (hunk 3 above)
Comment (hexqi) [reply .github/auto-labeler.yml:1]: steps里面用的是v4版本；不确定这个version的作用，文档里都是v1，先不改吧

**Review by chilingling — APPROVED** [review]

**Review by rhlin — APPROVED** [review]
good good

**Pull request merged**